Apps/TimelineApp/core/sensor/src/main/java/uk/ac/cam/cares/jps/sensor/source/handler/SoundLevelHandler.java

```diff
@@ -141,7 +159,7 @@ private void logDBFS(double dBFS) {
             dataPoint.put("time", System.currentTimeMillis() * 1000000);
             dataPoint.put("values", values);
 
-            synchronized (this) {
+            synchronized (sensorData) {
```

Comment:
it sounds like not a good idea to use sensorData as the lock, because it is not final.

And this should be a protection set in SensorHandlerManager as well, where it reads the sensorData. Consider to add this synchronization more properly with the other sensor handlers as well